lib/pushnotificationsservice.js

```diff
@@ -0,0 +1,350 @@
+      });
+    },
+    function(done) {
+      self.storage = new Storage();
+      self.storage.connect(opts.storageOpts, done);
+    },
+    function(done) {
+      self.messageBroker = opts.messageBroker || new MessageBroker(opts.messageBrokerOpts);
+      self.messageBroker.onMessage(_.bind(self._sendPushNotifications, self));
+      done();
+    },
+  ], function(err) {
+    if (err) {
+      log.error(err);
+    }
+    return cb(err);
+  });
+};
+
+PushNotificationsService.prototype._sendPushNotifications = function(notification, cb) {
+  var self = this;
+  var url = 'http://192.168.1.143:8000/send';
+  cb = cb || function() {};
+
+  var notifType = PUSHNOTIFICATIONS_TYPES[notification.type];
+  if (!notifType) return cb();
+
+  self._getRecipientsList(notification.walletId, function(err, recipientsList) {
+    self.storage.fetchWallet(notification.walletId, function(err, wallet) {
+
+      var resultedRecipientsList = _.reject(self._getJoinedRecipientsList(wallet, recipientsList), {
+        id: notification.creatorId || null
+      });
+      async.waterfall([
+        function(next) {
+          self._readAndApplyTemplates(notification, notifType, resultedRecipientsList, next);
+        },
+        function(contents, next) {
+          async.map(resultedRecipientsList, function(recipient, next) {
+            var opts = {};
+            var content = contents[recipient.language];
+            opts.users = [notification.walletId + '$' + recipient.id];
+            opts.android = {
+              "data": {
+                "title": content.plain.subject,
+                "message": content.plain.body
+              }
+            };
+            opts.ios = {
+              "badge": 0,
+              "alert": content.plain.body,
+              "sound": "soundName"
+            };
+            return next(err, opts);
+          }, next);
+        },
+        function(optsList, next) {
+          async.each(optsList,
+            function(opts, next) {
+              request({
+                url: url,
+                method: 'POST',
+                json: true,
+                body: opts
+              }, function(error, response, body) {
+                next();
+              });
+            },
+            function(err) {
+              log.error(err);
+              return cb(err);
+            }
+          );
+        },
+      ], function(err) {
+        if (err) {
+          log.error('An error ocurred generating notification', err);
+        }
+        return cb(err);
+      });
+    });
+  });
+};
+
+PushNotificationsService.prototype._getRecipientsList = function(walletId, cb) {
+  var self = this;
+
+  self.storage.fetchPreferences(walletId, null, function(err, preferences) {
+    if (err) return cb(err);
+    if (_.isEmpty(preferences)) return cb(null, []);
+
+    var recipients = _.compact(_.map(preferences, function(p) {
+
+      if (!_.contains(self.availableLanguages, p.language)) {
+        if (!p.language) {
```

Comment:
Isn't this condition somehow reversed? I think you would first check to see if the user prefers a specific language and if this is not the case you would assign `'en'`. Once this is done you would proceed to check available languages.